packages/apollo-cli/src/commands/codegen/__tests__/generate.test.ts

```diff
@@ -0,0 +1,146 @@
+jest.mock(
+  "apollo-codegen-core/lib/localfs",
+  () => {
+    return require("../../../__mocks__/localfs");
+  }
+);
+
+// this is because of herkou-cli-utils hacky mocking system on their console logger
+import { stdout, mockConsole } from "heroku-cli-util";
+import * as path from "path";
+import * as fs from "fs";
+import { test as setup } from "apollo-cli-test";
+import { introspectionQuery, print, execute, buildSchema } from "graphql";
+import gql from "graphql-tag";
+import { fs as mockFS, vol } from "apollo-codegen-core/lib/localfs";
+
+const test = setup.do(() => mockConsole());
+const fullSchema = execute(
+  buildSchema(
+    fs.readFileSync(path.resolve(__dirname, "../../schema/__tests__/fixtures/schema.graphql"), {
+      encoding: "utf-8",
+    })
+  ),
+  gql(introspectionQuery)
+).data;
+
+const simpleQuery = fs.readFileSync(path.resolve(__dirname, "./fixtures/simpleQuery.graphql"));
+
+beforeEach(() => {
+  vol.reset();
+  vol.fromJSON({
+    "__blankFileSoDirectoryExists": ""
+  });
+})
+
+jest.setTimeout(15000);
```

Comment:
was this timeout increase due to the new fs mocks?

Reply:
The tests were failing randomly, so I did this to make Travis more consistent.